packages/themes/ecl/src/ecl-ec/mixins/button.scss

```diff
@@ -1,102 +1,107 @@
-		outline-color: var(--color-white);
-	}
+			&:hover {
+				--text-background-color: var(--color-blue-130);
+				--text-border-color: var(--color-blue-130);
+			}
+		}
 
-	.primary a:hover .kol-span-wc,
-	.primary button:hover .kol-span-wc {
-		background-color: var(--color-blue-130);
-		border-color: var(--color-blue-130);
-	}
+		&--secondary {
+			--text-background-color: var(--color-white);
+			--text-border-color: var(--color-blue);
+			--text-color: var(--color-blue);
 
-	.secondary a .kol-span-wc,
-	.secondary button .kol-span-wc {
-		background-color: var(--color-white);
-		border-color: var(--color-blue);
-		color: var(--color-blue);
-	}
+			&:focus {
+				--text-focus-outline-color: var(--color-blue);
+				.kol-button__text {
+					outline-offset: rem(-6);
+				}
+			}
 
-	.secondary a:focus .kol-span-wc,
-	.secondary button:focus .kol-span-wc {
-		outline-color: var(--color-blue);
-		outline-offset: rem(-6);
-	}
+			&:hover {
+				--text-border-color: var(--color-blue-130);
+				--text-color: var(--color-blue-130);
+			}
+		}
 
-	.secondary a:hover .kol-span-wc,
-	.secondary button:hover .kol-span-wc {
-		border-color: var(--color-blue-130);
-		color: var(--color-blue-130);
-	}
+		&--normal {
+			--text-background-color: var(--color-yellow);
+			--text-border-color: var(--color-yellow);
+			--text-color: var(--color-black);
 
-	.normal a .kol-span-wc,
-	/* CTA */
-	.normal button .kol-span-wc {
-		background-color: var(--color-yellow);
-		border-color: var(--color-yellow);
-		color: var(--color-black);
-	}
+			&:hover {
+				--text-background-color: var(--color-yellow-120);
+				--text-border-color: var(--color-yellow-120);
+			}
+		}
 
-	.normal a:hover .kol-span-wc,
-	/* CTA */
-	.normal button:hover .kol-span-wc {
-		background-color: var(--color-yellow-120);
-		border-color: var(--color-yellow-120);
-	}
+		&--danger {
+			--text-background-color: var(--color-red);
+			--text-border-color: var(--color-red);
+			--text-color: var(--color-white);
 
-	.danger a .kol-span-wc,
-	.danger button .kol-span-wc {
-		background-color: var(--color-red);
-		border-color: var(--color-red);
-		color: var(--color-white);
-	}
+			&:focus {
+				--text-focus-outline-color: var(--color-white);
+			}
 
-	.danger a:hover .kol-span-wc,
-	.danger button:hover .kol-span-wc {
-		background-color: var(--color-red);
-		border-color: var(--color-red);
-	}
+			&:hover {
+				--text-background-color: var(--color-red);
+				--text-border-color: var(--color-red);
+			}
+		}
 
-	.ghost a .kol-span-wc,
-	.ghost button .kol-span-wc {
-		background-color: transparent;
-		border-color: transparent;
-		color: var(--color-blue);
-	}
+		&--ghost {
+			--text-background-color: transparent;
+			--text-border-color: transparent;
+			--text-color: var(--color-blue);
 
-	.ghost a:focus .kol-span-wc,
-	.ghost button:focus .kol-span-wc {
-		outline-color: var(--color-blue);
-	}
+			&:focus {
+				--text-focus-outline-color: var(--color-blue);
+				.kol-button__text {
```

Comment:
Das gleich hier, könnten wir diese Styles noch nach unten, in den eigentlichen __text-Block schieben?